text/0003-monorepo.md

```diff
@@ -0,0 +1,76 @@
+
+Pain points of keeping 73 repos up to date:
+
+* Lint rules always get out of date in some forgotten apps
+* Github templates have the same problem of lint rules
+* API patterns and best practices to develop a new block
+* Avoid duplication of logic
+* Test setup
+* Misleading open-source flow to create issues
+
+
+# Detailed design
+
+Since this is not a feature but an organizational change that might help everyone that works in Store Framework and the open-source community this we don't have much to add to this section, but we expect to fix all the problems listed above and that the monorepo help us to make it easier the following topics:
+
+* Share code to make testing easier
+* Create a calendar of support for majors
+* Plan new majors for the whole framework
+* Create migration guides to each change of these majors
+* Predictable manual updates to our clients to get a better store
+
+All these points about majors should have an RFC to explain better how we should do it.
+
+# Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- To launch a new major we would need to launch a major of all apps or to put the legacy major in a separate folder and to add the current in the place of the current major (similar with what the [search-result](https://github.com/vtex-apps/seach-result) does)
+- Older major versions that were being kept in different branches would have to make the same strategy to continue to exist
+- We would have to update the bot that launches a new version of the apps
+- The repository would be *incredibly* big
+
+# Alternatives
+
+If we have a tool that could make the best of two sides, creating a monorepo or using multiple repositories to each app. It would be something easy for the open-source community to adopt and easy for all of us that work on the Store Framework to communicate about API changes and to share code, but this would lead to a lot of work of VTEX too.
+
+# Adoption strategy
+
+We would have to adopt this in phases:
+
+1. Prepare the bot to launch new versions in this repository and prepare https://vtex.io/docs to find documentation in the repo
+2. Move at least two Store Framework repositories, setup all the tooling considering multiple apps, make sure that everything is working and that this has no implication in production (without sharing code until phase 6)
+3. Migrate all apps of the Store Framework team and stay this way for a while as a trial phase
+4. Migrate one team at a time to the monorepo
+5. Think about how we could share code between apps
```

Comment:
This might be a tricky one... since code would code from multiple apps. Maaaybe (big maybe 😂 ) we could even create an app for this shared code... Since all the code would be in the same repo, it doesn't seem like a horrible idea to me